python/paddle/v2/framework/tests/op_test.py

```diff
@@ -5,175 +5,313 @@
-                         input_to_check,
-                         output_names,
-                         delta=0.005,
-                         in_place=False):
-
-    set_input(scope, op, inputs, core.CPUPlace())
-    op.infer_shape(scope)
-
-    tensor_to_check = scope.find_var(input_to_check).get_tensor()
-
-    def product(dim):
+# TODO(pkuyym) simplify the code and add more comments
+class TestUtils(object):
+    @classmethod
+    def grad_var_name(cls, var_name):
+        return var_name + "@GRAD"
+
+    @classmethod
+    def get_tensor_by_name(cls, scope, var_name):
+        var = scope.find_var(var_name)
+        return var.get_tensor()
+
+    @classmethod
+    def create_variable(cls, scope, var_name, value=None, place=None):
+        var = scope.new_var(var_name)
+        if value is not None:
+            assert place is not None, \
+                'Place must be specified if value provided'
+            tensor = var.get_tensor()
+            tensor.set_dims(value.shape)
+            tensor.set(value, place)
+
+    @classmethod
+    def create_op(cls, scope, op_type, inputs, outputs, attrs):
+        kwargs = dict()
+        # parepare parameters for creating operator
+        for in_var_name, is_dup in Operator.get_op_inputs(op_type):
+            if in_var_name in inputs:
+                kwargs[in_var_name] = []
+                if is_dup:
+                    sub_vars = inputs[in_var_name]
+                    for sub_var_name, _ in sub_vars:
+                        cls.create_variable(scope, sub_var_name)
+                        kwargs[in_var_name].append(sub_var_name)
+                else:
+                    cls.create_variable(scope, in_var_name)
+                    kwargs[in_var_name].append(in_var_name)
+
+        for out_var_name, is_dup in Operator.get_op_outputs(op_type):
+            if out_var_name in outputs:
+                kwargs[out_var_name] = []
+                if is_dup:
+                    sub_vars = outputs[out_var_name]
+                    for sub_var_name, _ in sub_vars:
+                        cls.create_variable(scope, sub_var_name)
+                        kwargs[out_var_name].append(sub_var_name)
+                else:
+                    cls.create_variable(scope, out_var_name)
+                    kwargs[out_var_name].append(out_var_name)
+
+        for attr_name in Operator.get_op_attr_names(op_type):
+            if attr_name in attrs:
+                kwargs[attr_name] = attrs[attr_name]
+
+        return Operator(op_type, **kwargs)
+
+    @classmethod
+    def get_backward_op(cls, scope, op, no_grad_set):
+        backward_op = core.Operator.backward(op, no_grad_set)
+        for in_var_name in backward_op.input_vars():
+            cls.create_variable(scope, in_var_name)
+        for out_var_name in backward_op.output_vars():
+            cls.create_variable(scope, out_var_name)
+        return backward_op
+
+    @classmethod
+    def _feed_var(cls, scope, var_name, value, place):
```

Comment:
This function LGTM. But I'm not familiar with the underlining code. Review from someone else is required.